### DISCUSSION TITLE
How to combine vertical filtering and horizontal filtering ?

### DISCUSSION THREAD

**Discussion by Inrap** [PostgREST/postgrest#3521] [Q&A] [answered]
With a `PATCH` fetch, I would like to update a **specific field of a specific record** :
Let's say I only want to update the field named `alter` for the record for which `num = myvar`

I tried some stuff like : `http://.../t_fait?select=alter&num=eq.${myvar}` but obviously that not the correct way.
Thanks

**Comment by laurenceisla** [comment, marked as answer]
The `select` query parameter is used for [Vertical Filtering](https://postgrest.org/en/v12/references/api/tables_views.html#vertical-filtering), when using the `Prefer: return=representation` header. In other words, `select` filters the returned row(s) **after** the update is done.

What you need is the `columns` query parameter, which takes only the keys specified from the payload sent with the `PATCH` request. See [Specifying Columns](https://postgrest.org/en/v12/references/api/tables_views.html#specifying-columns) in the docs, it works the same for Inserts and Updates. In your case, replace select with columns, like this:

```
http://.../t_fait?columns=alter&num=eq.${myvar}
```

**Reply by laurenceisla** [reply]
As a side note, I noticed that the docs don't explicitly mention that both Specifying Columns and [using `Prefer: missing=default`](https://postgrest.org/en/v12/references/api/tables_views.html#bulk-insert-with-default-values) can be applied to Updates too. I'll open an issue for it.

Edit: opened issue #3522

**Comment by Inrap** [comment]
Thanks, now I know that `select` should to be used like a sql `returning`.
